src/ngraph/specialize_function.cpp

```diff
@@ -122,6 +129,7 @@ std::shared_ptr<Function>
     }
 
     auto function = std::make_shared<Function>(new_results, new_parameters);
+    function->set_leaf_nodes(new_assign_nodes);
```

Comment:
Added the list of leaf nodes to the class Lambda. Reason: Traversal functions (get_ops, get_ordered_ops) cut Assign operations from the graph because Assign ops often don't have output or Result. Traversals launched from Results do not find Assign ops.

Reply:
Does anything read the value after the assign? If so, this will not work because the read could be executed before the assign that it is supposed to be after.
You can use `add_control_dependency` to make the assigns need to happen before the first `Result` and that will cause them to be referenced and end up in the list of nodes without adding leaf nodes. In the case of a read after an assign, you would want to add a control dependency from the read to the assign to ensure things happened in the correct order. This can be very non-intuitive at the python level if there are python functions building sub-graphs that involve assignment.

Reply:
Thanks for your comment, `add_control_dependency `functionality will be used. This solves the problem.